packages/iTwinUI-react/src/core/Header/HeaderDropdownButton.tsx

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { DropdownMenu } from '../DropdownMenu';
+import SvgCaretDownSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretDownSmall';
+import SvgCaretUpSmall from '@itwin/itwinui-icons-react/cjs/icons/CaretUpSmall';
+import { PolymorphicForwardRefComponent, useTheme } from '../utils';
+import { DropdownButtonProps } from '../Buttons';
+import { useMergedRefs } from '../utils';
+
+export type HeaderDropdownButtonProps = DropdownButtonProps;
+type HeaderDropdownButtonComponent = PolymorphicForwardRefComponent<
+  'button',
+  HeaderDropdownButtonProps
+>;
+
+export const HeaderDropdownButton: HeaderDropdownButtonComponent = React.forwardRef(
+  (props, ref) => {
+    const { menuItems, className, size, children, startIcon, ...rest } = props;
+
+    useTheme();
+
+    const [isMenuOpen, setIsMenuOpen] = React.useState(false);
+
+    const [menuWidth, setMenuWidth] = React.useState(0);
+    const buttonRef = React.useRef<HTMLButtonElement>(null);
+    const refs = useMergedRefs(ref, buttonRef);
+
+    React.useEffect(() => {
+      if (buttonRef.current) {
+        setMenuWidth(buttonRef.current.offsetWidth);
+      }
+    }, [children, size]);
+
+    return (
+      <DropdownMenu
+        menuItems={menuItems}
+        style={{ minWidth: menuWidth }}
+        onShow={() => setIsMenuOpen(true)}
+        onHide={() => setIsMenuOpen(false)}
+      >
+        <button
```

Comment:
Why do you use `button` instead of `Button` or `HeaderBasicButton` here (and in `HeaderSplitButton`)? More control/fine-tuning?

Reply:
You are 100% correct, and this change solved a lot of the other inital issues raised in this PR. Thank you 😄